enhancements/ingress/lb-subnet-selection-aws.md

```diff
@@ -0,0 +1,600 @@
+the service and effectuates the subnet update.
+
+The `status.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets`
+will eventually reflect the configured subnet value by mirroring the value of
+`service.beta.kubernetes.io/aws-load-balancer-subnets` on the service. There are
+two scenarios in which the `Subnets` status won't be equal to the `Subnets` spec:
+
+1. The cluster admin manually configured the `service.beta.kubernetes.io/aws-load-balancer-subnets`
+   annotation on the service.
+2. The cluster admin configured the IngressController's `Subnets` spec, but hasn't
+   effectuated the change by deleting the service (see [Effectuating Subnet Updates](#effectuating-subnet-updates)).
+
+This proposal's use of `status.endpointPublishingStrategy` is consistent with the approach in
+[LoadBalancer Allowed Source Ranges](/enhancements/ingress/lb-allowed-source-ranges.md),
+but diverges with the approach in [Ingress Mutable Publishing Scope](/enhancements/ingress/mutable-publishing-scope.md).
+The Ingress Mutable Publishing Scope design sets `status.endpointPublishingStrategy.loadBalancer.scope`
+equal to `spec.endpointPublishingStrategy.loadBalancer.scope`, which may not always reflect the actual scope of the
+load balancer. This proposal for `Subnets` ensures `status.endpointPublishingStrategy` reflects the _actual_ value
+(in our case, the annotation value).
+
+### Workflow Description
+
+#### Creating a Custom IngressController with Subnets after Installation Workflow
+
+Creating an IngressController while specifying the subnets for the LoadBalancer-type
+service after installation:
+
+1. A cluster admin creates an IngressController with
+  `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` specified.
+2. The ingress operator creates the LoadBalancer-type Service and sets the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` annotation with the subnets
+   provided by the user.
+
+#### Updating an existing IngressController with new Subnets Workflow
+
+Updating an existing IngressController with new subnets requires the service
+to be deleted and is disruptive:
+
+1. A cluster admin edits an IngressController's
+   `spec.endpointPublishingStrategy.loadBalancer.providerParameters.aws.subnets` field.
+2. The Ingress Operator sets the `LoadBalancerProgressing` condition to `Status: True`
+   on the IngressController and adds status a message directing the cluster admin
+   to delete the service.
+3. The cluster admin deletes the LoadBalancer-type service indicated in the
+   `LoadBalancerProgressing` condition message.
+4. The Ingress Operator recreates the LoadBalancer-type service and configures the
+   `service.beta.kubernetes.io/aws-load-balancer-subnets` on the LoadBalancer-type
+   service, thus creating the load balancer with the desired subnets specified by the
+   cluster admin.
+
+#### Automatically Updating an existing IngressController with new Subnets Workflow
```

Comment:
nit: This should be referenced in the user stories section, like the other workflows.

Reply:
Good point - I have been linking a user story to workflows. Done.